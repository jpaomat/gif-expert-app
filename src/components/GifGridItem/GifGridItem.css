.card {
    align-content: center;
    border: 1px solid black;
    border-radius: 6px;
    height: 100px;
    margin: 5px;
    width: 100px;
    overflow: hidden;
}

img {
    height: 100px;
    width: 100px;
}