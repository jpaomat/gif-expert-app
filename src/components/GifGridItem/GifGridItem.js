import React from 'react';
import PropTypes from 'prop-types';
import './GifGridItem.css';

export const GifGridItem = ({url, title}) => {
    return (
        <div className='card animate__fadeOut'>
            <img
                src={url}
                alt={title}
            ></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
