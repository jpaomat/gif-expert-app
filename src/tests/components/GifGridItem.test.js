import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem/GifGridItem';

describe('Test to GifGridItem component', () => {
    let wrapper;
    let props;
    beforeEach(() => {
        props = {
            url: 'https://test',
            title: 'test'
        }
        wrapper = shallow(< GifGridItem {...props} />);
    })

    test('should create component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should validate the title is correct', () => {
        const paragraph = wrapper.find('p');
        expect(paragraph.text().trim()).toBe(props.title);
    })

    test('should get the img equal to url and alt of props', () => {
        const img = wrapper.find('img');
        const propsImg = img.props(); // obtiene todas las propiedades
        expect(propsImg.alt).toEqual(props.title); // img.prop('alt) -> propiedad especifica
        expect(propsImg.src).toEqual(props.url);
    })

    test('should have animate__fadeOut class', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeOut')).toBe(true);
    })


})
