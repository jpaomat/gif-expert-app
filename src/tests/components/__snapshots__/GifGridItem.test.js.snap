// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifGridItem component should create component 1`] = `
<div
  className="card animate__fadeOut"
>
  <img
    alt="test"
    src="https://test"
  />
  <p>
    test
  </p>
</div>
`;
